src/components/molecules/NavBar/NavBar.tsx

```diff
@@ -94,6 +87,12 @@ export const NavBar: React.FC<NavBarPropsType> = ({ hasBackButton = true }) => {
 
   const shouldShowHomeButton = hasSovereignVenue && !isSovereignVenue;
 
+  const {
+    isShown: isShowUserProfile,
+    show: showUserProfile,
+    hide: closeUserProfile,
+  } = useShowHide(false);
```

Comment:
Please, just reuse the `useProfileModalControls` hook to open the profile Modal

Reply:
Done. That also hinted me to merge `UserProfileModal` and `ProfileModal` into a single component. Commit https://github.com/sparkletown/sparkle/pull/1934/commits/3bd65c04013c7e4fa4fa98c71a008419c4ac0550